cmd/kube-proxy/app/server.go

```diff
@@ -801,11 +801,20 @@ func getConntrackMax(config kubeproxyconfig.KubeProxyConntrackConfiguration) (in
 	return 0, nil
 }
 
-// CleanupAndExit remove iptables rules and exit if success return nil
+// CleanupAndExit remove iptables rules and ipset/ipvs rules in ipvs proxy mode
+// and exit if success return nil
 func (s *ProxyServer) CleanupAndExit() error {
-	encounteredError := userspace.CleanupLeftovers(s.IptInterface)
-	encounteredError = iptables.CleanupLeftovers(s.IptInterface) || encounteredError
-	encounteredError = ipvs.CleanupLeftovers(s.IpvsInterface, s.IptInterface, s.IpsetInterface, s.CleanupIPVS) || encounteredError
+	// cleanup IPv6 and IPv4 iptables rules
+	ipts := []utiliptables.Interface{
+		utiliptables.New(s.execer, utiliptables.ProtocolIPv4),
+		utiliptables.New(s.execer, utiliptables.ProtocolIPv6),
+	}
+	var encounteredError bool
+	for _, ipt := range ipts {
+		encounteredError = userspace.CleanupLeftovers(ipt) || encounteredError
+		encounteredError = iptables.CleanupLeftovers(ipt) || encounteredError
+		encounteredError = ipvs.CleanupLeftovers(s.IpvsInterface, ipt, s.IpsetInterface, s.CleanupIPVS) || encounteredError
+	}
 	if encounteredError {
 		return errors.New("encountered an error while tearing down rules")
```

Comment:
So this will end up logging an error if `ip6tables` is uninstalled (even though there presumably weren't IPv6 iptables rules to clean up in that case). But that will need to be fixed to reconcile this with #94474 anyway.

If this is only going into 1.20 I'd rather hold this until #94474 and then rebase on top of that, but if this was going to be backported to 1.19 then it might be better to just merge this as-is and ignore the somewhat spurious error message in that case...

Reply:
I think cherry-pick to 1.19 is necessary, obviously 1.19 have this problem. 
But currently I have no idea about how to avoid spurious error message in this case, so maybe hold this until #94474 and then rebase on top of that still an option.